### ISSUE TITLE
Failing Oracle DB connection grabs loads of CPU's.

### ISSUE THREAD

**Issue by sjaakd** [quarkusio/quarkus#32454]
### Describe the bug

We observe across a number of Quarkus versions that a failing Oracle connection (either to wrong configuration or due to failing readiness checks ), Quarkus seems to get into a state that it grabs more and more CPUs.

We run a number of pod's and some of them we've got to restart on regular basis.

The problem seems to be related to the oracle  dependency.

```
        <dependency>
            <groupId>io.quarkus</groupId>
            <artifactId>quarkus-jdbc-oracle</artifactId>
        </dependency>
```

Note: we use Oracle Spatial dialect.

```
quarkus.hibernate-orm.dialect=org.hibernate.spatial.dialect.oracle.OracleSpatial10gDialect
```

We can provide a dump. 

We've seen this behavior already in a number of Quarkus versions, both in applications that use the "classic" as in the "reactive" suite. It seems (speculation) to trace its way back to the introduction of the above 'quarkus-jdbc-oracle' dependency.

For our classic applications  we use the following configuration:

```
quarkus.datasource.jdbc.new-connection-sql=SELECT 1 from dual
quarkus.datasource.jdbc.validation-query-sql=SELECT 1 from dual
``` 

### Expected behavior

gracefull connection shutdown and retry

### Actual behavior

increasing (when available) cpu consumption on our OpenShift cluster

### Output of `java -version`

openjdk 17.0.6 2023-01-17 LTS OpenJDK Runtime Environment (Red_Hat-17.0.6.0.10-3.el8_7) (build 17.0.6+10-LTS) OpenJDK 64-Bit Server VM (Red_Hat-17.0.6.0.10-3.el8_7) (build 17.0.6+10-LTS, mixed mode, sharing)

### Quarkus version or git rev

2.14.2.Final

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

Maven

### Additional information

4.10.53
Kubernetes version
v1.23.12+8a6bfe4

We can share a dump, but rather not via the ticket (not sure what information is available from that dump).

**Comment by geoand** [comment]
cc @Sanne

**Comment by geoand** [comment]
Does this same problem also occur with Quarkus `2.16.6.Final` and `3.0.0.CR2`?

**Comment by sjaakd** [comment]
We have found the issue.. (finally). It's in our own code. We use common interceptor (library) in all our Quarkus applications. One interceptor was executing:

```java
protected Throwable getCause(Throwable exception) {
    Throwable cause = exception;
    while( cause.getCause != null ) {
         cause = exception.getCause();
    } 
    return cause;
} 
```
causing an endless loop, only for some exceptions. We finally struck incidentally on a good reproducer (seen this problem since the we started using Quarkus and this common library is there from the start).

Thanks.
